packages/@aws-cdk/aws-kinesisanalyticsv2/lib/property-group.ts

```diff
@@ -0,0 +1,25 @@
+export interface PropertyMap {
+  [key: string]: string;
+}
+
+export interface CfnPropertyGroup {
+  readonly propertyGroupId: string;
+  readonly propertyMap: PropertyMap;
+}
+
+export class PropertyGroup {
+  private readonly propertyGroupId: string;
+  private readonly propertyMap: PropertyMap;
+
+  constructor(propertyGroupId: string, propertyMap: PropertyMap) {
+    this.propertyGroupId = propertyGroupId;
+    this.propertyMap = propertyMap;
+  }
+
+  public toCfn(): CfnPropertyGroup {
+    return {
+      propertyGroupId: this.propertyGroupId,
+      propertyMap: this.propertyMap,
+    };
+  }
+}
```

Comment:
I updated this to your suggested object literal style. I think I just got spooked by JSII and thought everything had to be classes rather than nested object literals.